test/detection/test_line_counter.py

```diff
@@ -70,3 +73,332 @@ def test_calculate_region_of_interest_limits(
+                [0, 50, 10, 40],
+            ],
+            [False, False, False],
+            [False, False, True],
+        ),
+        (
+            Vector(
+                Point(0, 0),
+                Point(0, 100),
+            ),
+            [
+                [100, 50, 120, 70],
+                [-100, 50, -80, 70],
+                [100, 50, 120, 70],
+                [-100, 50, -80, 70],
+                [100, 50, 120, 70],
+                [-100, 50, -80, 70],
+                [100, 50, 120, 70],
+                [-100, 50, -80, 70],
+            ],
+            [False, False, True, False, True, False, True, False],
+            [False, True, False, True, False, True, False, True],
+        ),
+        (
+            Vector(
+                Point(0, 0),
+                Point(-100, 0),
+            ),
+            [
+                [-50, 70, -40, 50],
+                [-50, -70, -40, -50],
+                [-50, 70, -40, 50],
+                [-50, -70, -40, -50],
+                [-50, 70, -40, 50],
+                [-50, -70, -40, -50],
+                [-50, 70, -40, 50],
+                [-50, -70, -40, -50],
+            ],
+            [False, False, True, False, True, False, True, False],
+            [False, True, False, True, False, True, False, True],
+        ),
+    ],
+)
+def test_line_zone_single_detection(
+    vector: Vector,
+    bbox_sequence: List[List[int]],
+    expected_crossed_in: List[bool],
+    expected_crossed_out: List[bool],
+) -> None:
+    line_zone = LineZone(start=vector.start, end=vector.end)
+    for i, bbox in enumerate(bbox_sequence):
+        detections = mock_detections(
+            xyxy=[bbox],
+            tracker_id=[0],
+        )
+        crossed_in, crossed_out = line_zone.trigger(detections)
+        assert crossed_in[0] == expected_crossed_in[i]
+        assert crossed_out[0] == expected_crossed_out[i]
+        assert line_zone.in_count == sum(expected_crossed_in[: (i + 1)])
+        assert line_zone.out_count == sum(expected_crossed_out[: (i + 1)])
+
+
+@pytest.mark.parametrize(
+    "vector,"
+    "bbox_sequence,"
+    "expected_crossed_in,"
+    "expected_crossed_out,"
+    "crossing_anchors",
+    [
+        (
+            Vector(
+                Point(0, 0),
+                Point(100, 100),
+            ),
+            [
+                [50, 30, 60, 20],
+                [20, 50, 40, 30],
+            ],
+            [False, False],
+            [False, True],
+            [Position.TOP_LEFT, Position.TOP_RIGHT, Position.BOTTOM_LEFT],
+        ),
+        (
+            Vector(
+                Point(0, 0),
+                Point(0, 100),
+            ),
+            [
+                [-100, 50, -80, 70],
+                [-100, 50, 120, 70],
+            ],
+            [False, True],
+            [False, False],
+            [Position.TOP_RIGHT, Position.BOTTOM_RIGHT],
+        ),
+    ],
+)
+def test_line_zone_single_detection_on_subset_of_anchors(
```

Comment:
@tc360950 or @LinasKo, could you explain a bit more about what we are testing here?

Reply:
@SkalskiP Other tests assume that the line is crossed with every anchor. This test checks LineZone behaviour when only a subset of anchors cross the line. For instance if only `[Position.BOTTOM_LEFT, Position.TOP_LEFT]` cross the line we check if 
LineZone returns correct results for (indicated as "crossing"):
`[Position.BOTTOM_LEFT, Position.TOP_LEFT]`
`[Position.TOP_LEFT]`
`[Position.BOTTOM_LEFT]`

and correct result for (indicated as not "crossing"):
`[Position.BOTTOM_LEFT, Position.TOP_LEFT, Position.TOP_RIGHT]`
`[Position.TOP_LEFT, Position.BOTTOM_RIGHT]`
`[Position.BOTTOM_RIGHT]`
etc. 

I've added clarifying docstring to every test.